src/ResourceTemplate/RelativeResourceTemplate.php

```diff
@@ -277,7 +331,8 @@ private static function flattenKeySegmentTuples(array $keySegmentTuples)
                 case Segment::VARIABLE_SEGMENT:
                     // For segment variables, replace the segment with the segments of its children
                     $template = $segment->getTemplate();
-                    $nestedKeySegmentTuples = self::buildKeySegmentTuples($template->segments);
+                    $nestedKeySegmentTuples =
```

Comment:
Just a minor style nit, but it if you are open to it, we generally prefer:

```php
$nestedKeySegmentTuples = self::buildKeySegmentTuples(
    $template->segments,
    $segment->getSeparator()
);
```

Reply:
Done.